airflow/models/mappedoperator.py

```diff
@@ -670,10 +675,12 @@ def prepare_for_execution(self) -> "MappedOperator":
         # we don't need to create a copy of the MappedOperator here.
         return self
 
+    @provide_session
     def render_template_fields(
         self,
         context: Context,
         jinja_env: Optional["jinja2.Environment"] = None,
+        session: Session = NEW_SESSION,
```

Comment:
We might not be able to do this because `render_template_fields` is public API that can be overriden. I think `@provide_session` can be put on `_resolve_expansion_kwargs` to avoid the problem.

Reply:
Done.